src/utils/components/Charts/NetworkUtil/NetworkThresholdChartSingleSource.tsx

```diff
@@ -28,14 +29,51 @@ const NetworkThresholdSingleSourceChart: React.FC<NetworkThresholdSingleSourceCh
   link,
 }) => {
   const { currentTime, duration, timespan } = useDuration();
-
   const { ref, width, height } = useResponsiveCharts();
 
-  const chartData = data?.map(([x, y]) => {
-    return { x: new Date(x * MILLISECONDS_MULTIPLIER), y: Number(y), name: 'Network In' };
+  const chartData =
+    !isEmpty(data) &&
+    data?.map((obj) => {
+      return (obj?.values || [])?.map(([x, y]) => {
+        return {
+          x: new Date(x * MILLISECONDS_MULTIPLIER),
+          y: Number(y),
+          name: obj?.metric?.interface,
+        };
+      });
+    });
+  const isReady = !isEmpty(chartData);
+
+  const findChartMaxYAxis = (chartData1: { x: Date; y: number; name: string }[][]) => {
+    const yValues =
+      !isEmpty(chartData1) &&
+      chartData1?.map((dataArray) => {
+        return Math.max(...dataArray?.map((data1) => data1?.y));
+      });
+    const maxY = Math.max(...(yValues || []));
+    return maxY;
+  };
+  const Ymax = findChartMaxYAxis(chartData);
+  const tickValues = Array.from({ length: Ymax + 1 }, (_, index) => {
+    if (index === 0) return '1 Bps';
+    if (index === Math.round(Ymax)) return `${Math.round(Ymax + 1)} Bps`;
+    return index.toString() + ' Bps';
   });
 
-  const isReady = !isEmpty(chartData);
+  const newTickFormat = (tick: any, index: number, ticks: any[]) => {
```

Comment:
can be exported to util file?
please consider changing the name to something more informative